### PR TITLE
#170764590 Avail requests for approval

### PR DESCRIPTION
**Pull request by Elvis-rugamba**
#### What does this PR do?
- Avail requests for approval
- Get all requests
- Get specific request
- Get pending requests
- Get approved requests
#### Description of Task to be completed?
- Create request controller
- create request model, migration and seeder
- create request service
- create request repository
- create request avail test
- create request schema
- create request middleware

#### How should this be manually tested?
- Clone this repo on your local machine, open your terminal and run npm i to install all dependencies
- start server npm run dev
- Use PATCH " api/requests/{request_id}/approve " to approve request
- Use GET " api/requests " to get all requests for super admin, to get all requests from his direct reports for manager, to get all requests created by a requester for requester
- Use GET " api/requests/{request_id}/ " to get specific request for super admin, to get specific request from his direct reports for manager, to get specific request created by a requester for requester
- Use GET " api/requests/status/{status}/ " to get all requests by status for super admin, to get all requests by status from his direct reports for manager, to get all requests by status created by a requester for requester
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170764590](https://www.pivotaltracker.com/story/show/170764590)

#### Screenshots (if appropriate)
N/A

#### Questions
None

### CODE REVIEW

**Review by Skemc — APPROVED** [review]
LGTM!

**Review by KarangwaEmmy — APPROVED** [review]
Cool

File: src/database/redis.database.js [hunk 1]
```diff
@@ -2,6 +2,6 @@ import dotenv from 'dotenv';
 import redis from 'redis';
 
 dotenv.config();
-const redisClient = redis.createClient(process.env.REDIS_URL);
```
Comment (ajimae) [inline src/database/redis.database.js:5]: Why did you remove the `REDIS_URL`?

**Review by ajimae — APPROVED** [review]
LGTM!
